test/integration/analytical_rosenbrock.cpp

```diff
@@ -12,102 +12,93 @@
+  test_analytical_stiff_ternary_chemical_activation(rosenbrock, 1e-4);
+  test_analytical_stiff_ternary_chemical_activation(backward_euler, 1e-3);
 }
 
 TEST(AnalyticalExamples, Tunneling)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_tunneling(builder);
+  test_analytical_tunneling(rosenbrock);
+  test_analytical_tunneling(backward_euler, 1e-3);
 }
 
 TEST(AnalyticalExamples, TunnelingSuperStiffButAnalytical)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_stiff_tunneling(builder);
+  test_analytical_stiff_tunneling(rosenbrock, 1e-4);
+  test_analytical_stiff_tunneling(backward_euler, 1e-3);
 }
 
 TEST(AnalyticalExamples, Arrhenius)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_arrhenius(builder);
+  test_analytical_arrhenius(rosenbrock);
+  test_analytical_arrhenius(backward_euler, 1e-3);
 }
 
 TEST(AnalyticalExamples, ArrheniusSuperStiffButAnalytical)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_stiff_arrhenius(builder);
+  test_analytical_stiff_arrhenius(rosenbrock, 1e-4);
+  test_analytical_stiff_arrhenius(backward_euler, 1e-3);
 }
 
 TEST(AnalyticalExamples, Branched)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_branched(builder);
+  test_analytical_branched(rosenbrock, 1e-3);
+  test_analytical_branched(backward_euler, 1e-1);
 }
 
 TEST(AnalyticalExamples, BranchedSuperStiffButAnalytical)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_stiff_branched(builder);
+  test_analytical_stiff_branched(rosenbrock, 1e-3);
+  test_analytical_stiff_branched(backward_euler, 1e-1);
 }
 
 TEST(AnalyticalExamples, Robertson)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_robertson(builder);
+  test_analytical_robertson(rosenbrock, 1e-1);
+  // test_analytical_robertson(backward_euler, 1e-1);
 }
 
 TEST(AnalyticalExamples, SurfaceRxn)
 {
-  auto builder = micm::CpuSolverBuilder<micm::RosenbrockSolverParameters>(
-      micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters());
-  test_analytical_surface_rxn(builder);
+  test_analytical_surface_rxn(rosenbrock, 1e-5);
+  test_analytical_surface_rxn(backward_euler, 1e-1);
```

Comment:
`1e-1` seems large to me. This makes me think whether we should write a function of calculating the relative error instead of using `expect_near`, which compares the absolute difference.

Reply:
yeah these values are on the small side. I'll make that change

Reply:
done, only was able to use 0.05 for the relative error though